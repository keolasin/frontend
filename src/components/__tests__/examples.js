import React from "react"
import { render, cleanup } from "@testing-library/react"
import styled from "@emotion/styled"

afterEach(cleanup);

// test component for react-testing-library
const Title = () => {
  return (
    <h1 data-testid="hero-title">Gatsby is awesome!</h1>
  )
}

// test button for emotion snapshot
const Button = styled.div`
  color: hotpink;
`;


// test suite for react component
test("Test title component displays the correct title", () => {
  // Arrange
  const { getByTestId } = render(<Title />);

  // Act

  // Assertion
  expect(
    getByTestId("hero-title")
  ).toHaveTextContent("Gatsby is awesome!");
})

// test suite for snapshot
test("Emotion styled button renders correctly", () => {
  // Arrange
  const { asFragment } = render(<Button>This is hotpink.</Button>);

  // Act

  // Assertion
  expect(
    asFragment()
  ).toMatchSnapshot()
})