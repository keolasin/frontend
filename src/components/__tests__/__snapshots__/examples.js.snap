// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Emotion styled button renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  color: hotpink;
}

<div
    class="emotion-0 emotion-1"
  >
    This is hotpink.
  </div>
</DocumentFragment>
`;
